src/python/txtai/workflow/task/storage.py

```diff
@@ -107,11 +109,7 @@ def list(self, element):
         client = driver(
             key,
             secret,
-            host=self.host,
-            port=self.port,
-            token=self.token,
-            region=self.region,
-        )
+            **{field: self.__dict__.get(field) for field in ["secure", "host", "port", "api_version", "region", "token"] if self.__dict__.get(field)})
```

Comment:
Same comment as above on `secure`. Seems like something we wouldn't want to give the option on to prevent ill advised decisions.